### PR TITLE
Add Laracasts search as extension

### PR DESCRIPTION
**Pull request by danyelkeddah**
## Description

Add [Laracasts](https://laracasts.com/) search as extension to raycast.

## Screencast

raycast-laracasts

## Checklist

- [X] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [X] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [X] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

File: extensions/laracasts/package.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  "name": "laracasts",
+  "title": "Laracasts",
+  "description": "Laracasts Search",
+  "icon": "command-icon.png",
+  "author": "danyelkeddah",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "laracasts",
+      "title": "Search Laracasts",
```
Comment (pernielsentikaer) [inline extensions/laracasts/package.json:12]:
```suggestion
      "title": "Search Course",
```
Could that go?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @danyelkeddah 👋 

Thanks for another contribution 💯 

I have some feedback ready for you to address 💪 

- Could you move the screenshot to `media` folder

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

File: extensions/laracasts/src/laracasts.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  const algoliaClient = useMemo(() => {
+    return algoliaSearch(APPID, APIKEY);
+  }, [APPID, APIKEY]);
+
+  const algoliaIndex = useMemo(() => {
+    return algoliaClient.initIndex(INDEX);
+  }, [algoliaClient, INDEX]);
+
+  const [searchResults, setSearchResults] = useState<any[] | undefined>();
+  const [isLoading, setIsLoading] = useState(false);
```
Comment (pernielsentikaer) [inline extensions/laracasts/src/laracasts.tsx:62]:
```suggestion
  const [isLoading, setIsLoading] = useState(true);
```
We need this to remove [empty state flicker](https://developers.raycast.com/basics/prepare-an-extension-for-store#empty-states)

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @danyelkeddah 👋 

Just found one more to be corrected, when this is done, then we're ready to take it to the next level 💯 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by danyelkeddah** [comment]
Thanks @pernielsentikaer 👍🏻 , Just applied the fixes and ready for your review.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @danyelkeddah 👋 

Thanks for the update, it does however seem like there is a bit more [empty state flicker](https://developers.raycast.com/basics/prepare-an-extension-for-store#empty-states) that we didn't catch!

When logging, I see this:

```
07:06:21.107 isLoading: true | items found: undefined
07:06:21.517 isLoading: false | items found: undefined
07:06:21.519 isLoading: false | items found: 1
```

Seems like it still stops the loader before it's ready, can you address this?

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Review by pernielsentikaer — APPROVED** [review]
Awesome @danyelkeddah 💪 Much appreciated 💯 

I'm going to approve it, well done 🚀

**Pull request merged**